File: app/src/main/java/com/example/dev/gymassistantv2/Stopwatch.java
package com.example.dev.gymassistantv2;

/**
 * Created by Dev on 08.02.2018.
 */

public class Stopwatch {

    private long startTime = 0;
    private long stopTime = 0;
    private boolean running = false;


    public void start() {
        if(startTime == 0) {
            startTime = System.currentTimeMillis();
        }
        else if(!running) {
            startTime = System.currentTimeMillis() - (stopTime - startTime);
        }
        running = true;
    }


    public void stop() {
        if(running) {
            stopTime = System.currentTimeMillis();
        }
        running = false;
    }

    public void reset() {
        startTime = 0;
        stopTime = 0;
        running = false;
    }

    public void set(long elapsedTime) {
        startTime = System.currentTimeMillis() - elapsedTime;
        stopTime = System.currentTimeMillis();
    }

    // elaspsed time in milliseconds
    public long getElapsedTime() {
        if (running) {
            return System.currentTimeMillis() - startTime;
        }
        return stopTime - startTime;
    }

    public String getFormattedElapsedTime() {

        String minutes = Long.toString(getElapsedTime() / 60000 % 60);
        if(minutes.length() < 2) {
            minutes = "0" + minutes;
        }

        String seconds = Long.toString(getElapsedTime() / 1000 % 60);
        if(seconds.length() < 2) {
            seconds = "0" + seconds;
        }

        String hundredths = Long.toString(getElapsedTime() / 10 % 100);
        if(hundredths.length() < 2) {
            hundredths = "0" + hundredths;
        }

        return minutes + ":" + seconds + ":" + hundredths;
    }

    public boolean isRunning() {
        return running;
    }
}

File: app/src/main/java/com/example/dev/gymassistantv2/utils/Stopwatch.kt
package com.example.dev.gymassistantv2.utils

class Stopwatch {

    private var startTime: Long = 0
    private var stopTime: Long = 0
    var isRunning = false

    val elapsedTime: Long
        get() = if (isRunning) {
            System.currentTimeMillis() - startTime
        } else stopTime - startTime

    val formattedElapsedTime: String
        get() {

            var minutes = java.lang.Long.toString(elapsedTime / 60000 % 60)
            if (minutes.length < 2) {
                minutes = "0$minutes"
            }

            var seconds = java.lang.Long.toString(elapsedTime / 1000 % 60)
            if (seconds.length < 2) {
                seconds = "0$seconds"
            }

            var hundredths = java.lang.Long.toString(elapsedTime / 10 % 100)
            if (hundredths.length < 2) {
                hundredths = "0$hundredths"
            }

            return "$minutes:$seconds:$hundredths"
        }


    fun start() {
        if (startTime == 0L) {
            startTime = System.currentTimeMillis()
        } else if (!isRunning) {
            startTime = System.currentTimeMillis() - (stopTime - startTime)
        }
        isRunning = true
    }


    fun stop() {
        if (isRunning) {
            stopTime = System.currentTimeMillis()
        }
        isRunning = false
    }

    fun reset() {
        startTime = 0
        stopTime = 0
        isRunning = false
    }

    fun set(elapsedTime: Long) {
        startTime = System.currentTimeMillis() - elapsedTime
        stopTime = System.currentTimeMillis()
    }
}